pkg/utils/kubernetes/reconciler.go

```diff
@@ -0,0 +1,120 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"github.com/RHsyseng/operator-utils/pkg/resource"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+type Finalizer interface {
+	GetName() string
+	OnFinalize(owner resource.KubernetesResource, service PlatformService) error
+}
+
+type ExtendedReconciler struct {
```

Comment:
FinalizingReconciler? Why renamed back to reconciler.go? Do these classes have to do with anything other than finalizers?

Reply:
The approach taken here is different and although I am only introducing a wrapper for finalizers this solution opens the door for multiple extensions around the base Reconciler.
What about before/afterReconcile listeners or filters?

Reply:
Yeah, it sounds great to have before/after filters. You can then have the filter interface and the ExtendedReconciler in this reconciler.go file, but move everything else including the Finalizer filter implementation to a finalizer.go file or something.

Reply:
It is all part of the ExtendedReconciler logic. I don't see the benefit of moving the Finalizer interface and the ExtendedReconciler methods related to finalizers to a separate file as they still belong to the reconciler struct. IMO it makes more sense to have it together.

Reply:
As discussed, I'd like to see this better organized.

Reply:
Done :+1: